### PR TITLE
Unit tests

### PR DESCRIPTION
**Pull request by ayntee**
# Unit tests

**Note: some of the functions are premium-only features (by Discord) or utilize a core functions―helper functions; therefore, may not be included in the unit tests.**

Closes: #106

### CODE REVIEW

**Comment by Skillz4Killz** [comment]
Let's move the checklist on here to separate issues for anything not yet completed.

File: tests/mod.test.ts [hunk 1]
```diff
@@ -0,0 +1,245 @@
+import { assert, assertArrayContains, assertEquals, delay } from "../deps.ts";
+import {
+  botID,
+
```
Comment (Skillz4Killz) [inline tests/mod.test.ts:4]: Can u look into these empty spaces

File: tests/mod.test.ts [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+  OverwriteType,
+  Role,
+
+  sendMessage,
+} from "../mod.ts";
+import { editChannel } from "../src/handlers/channel.ts";
+import { getChannel } from "../src/handlers/guild.ts";
+
+const token = Deno.env.get("DISCORD_TOKEN");
```
Comment (Skillz4Killz) [inline tests/mod.test.ts:33]: Need to create a reminder to make this for the action

File: tests/mod.test.ts [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+    // Delay the execution by 15 seconds (15000 ms)
+    await delay(15000);
+
+    // Check whether botID is nil or not
+    assert(botID);
+  },
+  ...testOptions,
+});
+
+let guildID: string;
```
Comment (Skillz4Killz) [inline tests/mod.test.ts:59]: For long term wise we should move all these vars into 1 placeholder at the top like

```ts
const data = {
  guildID: "",
   roleID: ""
};
```

Then we can assign and delete by data.guildID = x

Comment (Skillz4Killz) [reply tests/mod.test.ts:59]: It would save us from having to createa ton of vars

File: tests/mod.test.ts [hunk 4]
```diff
@@ -0,0 +1,245 @@
...
+    channelID = channel.id;
+  },
+});
+
+Deno.test({
+  name: "channel overwrite has permission",
+  async fn() {
+    const channel = cache.channels.get(channelID);
+    if (!channel) throw "Channel not found";
+    assertArrayContains(channel.permission_overwrites!, [
```
Comment (Skillz4Killz) [inline tests/mod.test.ts:166]: Add this in a separate issue for v10 to fix this snake case. Anything u see in snake case should be fixed in a big major breaking change

File: tests/mod.test.ts (hunk 2 above)
Comment (Skillz4Killz) [reply tests/mod.test.ts:33]: Maybe new issue for now easy to remind me when im free to do this

**Comment by Skillz4Killz** [comment]
I would also like to see a badge on the readme somewhere about this, we can keep it at the bottom of the readme until we fix this properly for tests to succeeed

**Comment by chroventer** [comment]
> Let's move the checklist on here to separate issues for anything not yet completed.

I have removed the checklist from here and posted it as a comment in the main issue.

**Comment by chroventer** [comment]
> I would also like to see a badge on the readme somewhere about this, we can keep it at the bottom of the readme until we fix this properly for tests to succeeed

Done :)

**Pull request merged**